### PR TITLE
Update OpenLayers --> v7.5.0

### PR DESCRIPTION
**Pull request by spwoodcock**
- ~~Anticipates the merge of https://github.com/openlayers-elements/openlayers-elements/pull/107~~
- Updates to latest OpenLayers v7 version.
- Minor fixes:
  - Update some of the typing for the ol-marker-icon component.
  - Replace the deprecated `autoPanAnimation` from Overlay, in favour of using the `animation` property of `Overlay`: `Overlay({animation: true})`.
  - Fixes tests for ol-overlay.

### CODE REVIEW

**Review by ludovicm67 — CHANGES_REQUESTED** [review]
Thank you @spwoodcock for your work! :)

I run the following command on that branch:

```sh
npm install
npm run build
npm run start
```

It's able to build, but in the logs of the `npm run start` it looks like there is one error among all successful logs:

```
|     ERROR in ../node_modules/ol/proj/proj4.js 160:18
 |     Module parse failed: Unexpected token (160:18)
 |     File was processed with these loaders:
 |      * ../node_modules/babel-loader/lib/index.js
 |     You may need an additional loader to handle the result of these loaders.
 |     |     return response.json().then(json => {
 |     |       const results = json['results'];
 |     >       if (results?.length > 0) {
 |     |         const result = results.filter(r => r['id']?.['authority'] === 'EPSG' && r['id']?.['code'] === code)[0];
 |     |         if (result) {
 |      @ ../elements/swisstopo-elements/projections.js 1:0-44 5:0-8
 |      @ ../elements/swisstopo-elements/swisstopo-wmts.js
 |      @ ./demo/swiss-topo/index.js
```

And this demo seems to be broken (no maps are shown on the page anymore): http://0.0.0.0:8080/#/elements/swisstopo-wmts/demos/demo/swiss-topo/
It was working with v6.

Can you have a look?

**Comment by spwoodcock** [comment]
> Thank you @spwoodcock for your work! :)
> 
> I run the following command on that branch:
> 
> ```shell
> npm install
> npm run build
> npm run start
> ```
> 
> It's able to build, but in the logs of the `npm run start` it looks like there is one error among all successful logs:
> 
> ```
> |     ERROR in ../node_modules/ol/proj/proj4.js 160:18
>  |     Module parse failed: Unexpected token (160:18)
>  |     File was processed with these loaders:
>  |      * ../node_modules/babel-loader/lib/index.js
>  |     You may need an additional loader to handle the result of these loaders.
>  |     |     return response.json().then(json => {
>  |     |       const results = json['results'];
>  |     >       if (results?.length > 0) {
>  |     |         const result = results.filter(r => r['id']?.['authority'] === 'EPSG' && r['id']?.['code'] === code)[0];
>  |     |         if (result) {
>  |      @ ../elements/swisstopo-elements/projections.js 1:0-44 5:0-8
>  |      @ ../elements/swisstopo-elements/swisstopo-wmts.js
>  |      @ ./demo/swiss-topo/index.js
> ```
> 
> And this demo seems to be broken (no maps are shown on the page anymore): http://0.0.0.0:8080/#/elements/swisstopo-wmts/demos/demo/swiss-topo/ It was working with v6.
> 
> Can you have a look?

Definitely! I made these commits quite a few months ago now & I wonder what I missed 🤦

**Comment by tpluscode** [comment]
Let's continue here once we have #117 merged

**Comment by spwoodcock** [comment]
I just pushed the fix - I had to:
- Update some of the typing for the `ol-marker-icon` component.
- Replace the deprecated **autoPanAnimation** from `ol-overlay`, in favour of using the `animation` property of `autoPan`.

Everything seems to be working for `npm run build` and the overlay demo in `npm run start` 👍

**Review by ludovicm67 — COMMENTED** [review]
Looks good, but some tests seem to be failing.
Maybe they need to be updated? Or something needs to be fixed?

File: elements/openlayers-elements/ol-overlay.ts [hunk 1]
```diff
@@ -77,19 +77,32 @@ export default class OlOverlay extends OlMapPart<Overlay> {
...
 
-    this.__overlay = new Overlay({
+    const overlayOptions: {
+      element: HTMLElement
+      autoPan?: {
+        animation: {
+          duration: number
+        }
+      }
+    } = {
```
Comment (tpluscode) [inline elements/openlayers-elements/ol-overlay.ts:87]: This type is exported so you could `import { Options } from 'ol/Overlay.js'` and use it to annotate

```suggestion
    const overlayOptions: Options = {
```

**Review by tpluscode — CHANGES_REQUESTED** [review]
Please also add a changeset by running `npx changeset` or doing so from the bot PR comment

**Comment by spwoodcock** [comment]
Thanks guys - I'll remember both tests and changeset in future PRs 😁

File: elements/openlayers-elements/ol-overlay.ts [hunk 2]
```diff
@@ -79,16 +80,12 @@ export default class OlOverlay extends OlMapPart<Overlay> {
 
     const overlayOptions: {
       element: HTMLElement
-      autoPan?: {
-        animation: {
-          duration: number
-        }
-      }
+      autoPan?: PanIntoViewOptions | boolean
```
Comment (tpluscode) [inline elements/openlayers-elements/ol-overlay.ts:83]: 👍

File: elements/openlayers-elements/test/ol-overlay.test.ts [hunk 3]
```diff
@@ -15,7 +15,10 @@ describe('ol-overlay', () => {
       const overlay = await element.createPart()
 
       // then
-      expect(overlay.getOptions().autoPan).to.equal(true)
+      const autoPanOptions = overlay.getOptions().autoPan
+      if (typeof autoPanOptions === 'object') {
+        expect(autoPanOptions.animation?.duration).to.equal(0)
+      }
```
Comment (tpluscode) [inline elements/openlayers-elements/test/ol-overlay.test.ts:21]: You didn't like `deep.equals`? Saves you from the conditional

```suggestion
      expect(overlay.getOptions().autoPan).to.deep.equal({ animation: { duration: 0 } })
```

**Review by tpluscode — APPROVED** [review]
Optional suggestion in tests. Otherwise 👍

File: elements/openlayers-elements/test/ol-overlay.test.ts (hunk 3 above)
Comment (spwoodcock) [reply elements/openlayers-elements/test/ol-overlay.test.ts:21]: Much nicer 👍 Thanks for the pointer!

**Pull request merged**